pkg/relayer/session/proof.go

```diff
@@ -258,6 +281,94 @@ func (rs *relayerSessionsManager) goProveClaims(
 		successProofs = append(successProofs, sessionTree)
 	}
 
-	failSubmitProofsSessionsCh <- failedProofs
+	failGenerateProofsSessionsCh <- failedProofs
 	proofsGeneratedCh <- successProofs
 }
+
+// isProofRequired determines whether a proof is required for the given session's
+// claim based on the current proof module governance parameters.
+// TODO_TECHDEBT: Once the on/off-chain loggers are unified, move this logic to
+// a shared helper used by both off-chain and on-chain routines.
+func (rs *relayerSessionsManager) isProofRequired(
+	ctx context.Context,
+	sessionTree relayer.SessionTree,
+	proofPathSeedBlock client.Block,
+) (isProofRequired bool, err error) {
+	logger := rs.logger.With(
+		"session_id", sessionTree.GetSessionHeader().GetSessionId(),
+		"claim_root", fmt.Sprintf("%x", sessionTree.GetClaimRoot()),
+		"supplier_operator_address", sessionTree.GetSupplierOperatorAddress().String(),
+	)
+
+	// Create the claim object and use its methods to determine if a proof is required.
+	claim := claimFromSessionTree(sessionTree)
+
+	// Get the number of compute units accumulated through the given session.
+	numClaimComputeUnits, err := claim.GetNumComputeUnits()
+	if err != nil {
+		return false, err
+	}
+
+	proofParams, err := rs.proofQueryClient.GetParams(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	logger = logger.With(
+		"num_claim_compute_units", numClaimComputeUnits,
+		"proof_requirement_threshold", proofParams.GetProofRequirementThreshold(),
+	)
+
+	// Require a proof if the claim's compute units meets or exceeds the threshold.
+	if numClaimComputeUnits >= proofParams.GetProofRequirementThreshold() {
+		logger.Info().Msg("compute units is above threshold, claim requires proof")
+
+		return true, nil
+	}
+
+	// Get the hash of the claim to seed the random number generator.
+	var claimHash []byte
+	claimHash, err = claim.GetHash()
+	if err != nil {
+		return false, err
+	}
+
+	// Append the hash of the proofPathSeedBlock to the claim hash to seed the random number generator
+	// to ensure that the proof requirement probability is unknown until the proofPathSeedBlock is observed.
+	// The on-chain claim settlement routine will use the same seed to determine if a proof is required.
+	proofRequirementSeed := append(claimHash, proofPathSeedBlock.Hash()...)
+
+	// Sample a pseudo-random value between 0 and 1 to determine if a proof is required probabilistically.
+	var randFloat float32
+	randFloat, err = poktrand.SeededFloat32(proofRequirementSeed)
```

Comment:
Don't like this `randFloat` naming. See my other comment and reuse the helper that I recommend creating from the onchain code.

Reply:
Renaming to `proofRequirementCheckValue` so it's distinguishable from the gov param. Here's how it'll show in logs:
```go
logger = logger.With(
	"proof_requirement_check_value", proofRequirementCheckValue,
	"proof_request_probability", proofParams.GetProofRequestProbability(),
)
```